components/org.wso2.carbon.identity.auth.abstractotp/src/main/java/org/wso2/carbon/identity/auth/abstractotp/internal/AuthenticatorDataHolder.java

```diff
@@ -0,0 +1,142 @@
+package org.wso2.carbon.identity.auth.abstractotp.internal;
+
+import org.wso2.carbon.identity.event.services.IdentityEventService;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.handler.event.account.lock.service.AccountLockService;
+import org.wso2.carbon.idp.mgt.IdpManager;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * Encapsulates the data of OTP authenticator.
+ */
+public class AuthenticatorDataHolder {
+
+    private static volatile AuthenticatorDataHolder authenticatorDataHolder = new AuthenticatorDataHolder();
+
+    private static RealmService realmService;
+    private static AccountLockService accountLockService;
+    private static IdentityGovernanceService identityGovernanceService;
+    private static IdentityEventService identityEventService;
+    private static IdpManager idpManager;
+
+    private AuthenticatorDataHolder() {
+
+    }
+
+    public static AuthenticatorDataHolder getInstance() {
+
+        return authenticatorDataHolder;
+    }
+
+    /**
+     * Get the RealmService.
+     *
+     * @return RealmService.
+     */
+    public static RealmService getRealmService() {
+
+        if (realmService == null) {
```

Comment:
Throwing a RuntimeException does not provide any value IMO.

Reply:
Removed the null check and the RuntimeException